iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -4,1310 +4,1097 @@
-            // after running for more than 24 hours
-            try
+            if (disconnectedEventArgs.ClientWasConnected)
             {
-                if ((State & TransportState.Open) == TransportState.Open)
-                {
-                    string topic = message.MqttTopicName;
-                    if (Logging.IsEnabled)
-                        Logging.Info(this, $"Received a message on topic: {topic}", nameof(OnMessageReceived));
-
-                    if (topic.StartsWith(TwinResponseTopicPrefix, StringComparison.OrdinalIgnoreCase))
-                    {
-                        _twinResponseEvent(message);
-                    }
-                    else if (topic.StartsWith(TwinPatchTopicPrefix, StringComparison.OrdinalIgnoreCase))
-                    {
-                        await HandleIncomingTwinPatchAsync(message).ConfigureAwait(false);
-                    }
-                    else if (topic.StartsWith(MethodPostTopicPrefix, StringComparison.OrdinalIgnoreCase))
-                    {
-                        await HandleIncomingMethodPostAsync(message).ConfigureAwait(false);
-                    }
-                    else if (topic.StartsWith(_receiveEventMessagePrefix, StringComparison.OrdinalIgnoreCase))
-                    {
-                        await HandleIncomingEventMessageAsync(message).ConfigureAwait(false);
-                    }
-                    else if (topic.StartsWith(_deviceboundMessagePrefix, StringComparison.OrdinalIgnoreCase))
-                    {
-                        _messageQueue.Enqueue(message);
-                        if (_isDeviceReceiveMessageCallbackSet)
-                        {
-                            await HandleIncomingMessagesAsync().ConfigureAwait(false);
-                        }
-                        else
-                        {
-                            _receivingSemaphore.Release();
-                        }
-                    }
-                    else
-                    {
-                        if (Logging.IsEnabled)
-                            Logging.Error(this, $"Received MQTT message on an unrecognized topic, ignoring message. Topic: {topic}");
-                    }
-                }
+                OnTransportDisconnected();
             }
-            catch (Exception ex)
-            {
-                if (Logging.IsEnabled)
-                    Logging.Error(this, $"Received an exception while processing an MQTT message: {ex}", nameof(OnMessageReceived));
 
-                OnError(ex);
-            }
-            finally
-            {
-                if (Logging.IsEnabled)
-                    Logging.Exit(this, message, nameof(OnMessageReceived));
-            }
+            return Task.CompletedTask;
         }
 
-        private async Task HandleIncomingEventMessageAsync(Message message)
+        private async Task HandleReceivedMessage(MqttApplicationMessageReceivedEventArgs receivedEventArgs)
```

Comment:
If the method is Async, should it then be titled `HandleRecievedMessageAsync()`?